server/controllers/answer.js

```diff
@@ -1,152 +1,221 @@
    * @param {object} res Response
-   * @returns {object} Accepted answer object or error object if answer is not found
+   * @returns {object} Updated answer object
+   * or error object if answer is not found
    */
-  static acceptAnswer(req, res) {
+  static updateAnswer(req, res) {
     const { questionId, answerId } = req.params;
-    const { userId } = req.body;
-
-    const user = userRepo.getUser(userId);
-    if (!user) {
-      return errors.notFound(res, 'user');
-    }
-
-    const answer = answerRepo.getAnswer(answerId, questionId);
-    if (!answer) {
-      return errors.notFound(res, 'answer');
-    }
-
-    const question = repo.getQuestion(questionId);
-    if (!question) {
-      return errors.notFound(res, 'question');
-    }
-
-    if (question.user.id !== userId) {
-      return errors.unauthorized(res);
-    }
-
-    let acceptedAnswer = question.answers.find(ans => ans.isAccepted);
-    if (acceptedAnswer) {
-      return errors.unauthorized(res);
-    }
+    const { userId, answer } = req.body;
 
-    acceptedAnswer = answerRepo.acceptAnswer(answerId);
-    return res.status(200).json({
-      status: 'success',
-      message: 'your have accepted this answer',
-      acceptedAnswer,
-    });
+    pool.connect()
+      .then((client) => {
+        client.release();
+        client.query(questionQueries.getQuestion(questionId))
+          .then((response) => {
+            const [questionObj] = response.rows;
+            if (!questionObj) {
+              return errors.notFound(res, 'question');
+            }
+            client.query(answerQueries.getAnswer(answerId))
+              .then((ansResponse) => {
+                const [existingAnswer] = ansResponse.rows;
+                if (!existingAnswer) {
+                  return errors.notFound(res, 'answer');
+                }
+                if (existingAnswer.questionid !== questionId) {
+                  return res.status(404).json({
+                    status: 'error',
+                    message:
+                    'Bad request. This answer belongs to another question.',
+                  });
+                }
+                if (userId !== existingAnswer.userid
+                  && userId !== questionObj.userid) {
+                  return errors.unauthorized(res);
+                }
+                if (existingAnswer.userid === userId) {
+                  if (answer) {
+                    client.query(answerQueries
+                      .updateAnswer(answerId, 'answer', answer))
+                      .then((updateRes) => {
+                        const [updatedAnswer] = updateRes.rows;
+                        return res.status(200).json({
+                          status: 'success',
+                          message: 'you have updated your answer',
+                          updatedAnswer,
+                        });
+                      });
+                  } else if (questionObj.userid !== userId) {
```

Comment:
- This kind of checks should be abstracted to a middleware. We do not want our controller to be doing a task on unchecked parameters
-